### PR TITLE
Update recipe for py312_patch

### PR DESCRIPTION
**Pull request by cbouss**
Preparation for py312 release.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,14 +10,13 @@ source:
...
-  number: 0
+  number: 1
   skip: True  # [py<37]
   script: {{ PYTHON }} -m pip install . -vv
 
 requirements:
   host:
-    # Cannot use flit-core >3.3.0 due to cyclic dependency 
-    - flit-core 3.3.0
+    - flit-core 3
```
Comment (ryanskeith) [inline recipe/meta.yaml]: Just a caution that upstream wants 3.2.0 or greater.

**Review by ryanskeith — APPROVED** [review]
LGTM.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,14 +10,13 @@ source:
   sha256: de526c12914f0c550d15924c62d72abc48d6fe7364aa87328337a31007fe8a4f
 
 build:
-  number: 0
+  number: 1
   skip: True  # [py<37]
   script: {{ PYTHON }} -m pip install . -vv
```
Comment (ryanskeith) [inline recipe/meta.yaml]:
```suggestion
  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
```

**Review by ryanskeith — COMMENTED** [review]
Some tidying.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Pull request merged**